test/test-url.cpp

```diff
@@ -1154,3 +1196,39 @@ TEST(PunycodeTest, Safe)
     std::string encoded("http://safe.segments/");
     EXPECT_EQ(encoded, Url::Url(unencoded).punycode().str());
 }
+
+TEST(PunycodeTest, SingleDot)
+{
+    ASSERT_THROW(Url::Url("http://./").punycode(), std::invalid_argument);
```

Comment:
Thanks for adding this case. Was it handled before or did you have to do something to handle this case?